contracts/service-registry/src/contract.rs

```diff
@@ -3172,4 +3172,138 @@ mod test {
+        let chain_name = ChainName::from_str("ethereum").unwrap();
+
+        // Bond and register all verifiers at once
+        for verifier in &original_verifiers {
+            let res = execute(
+                deps.as_mut(),
+                mock_env(),
+                message_info(
+                    &api.addr_validate(verifier).unwrap(),
+                    &coins(min_verifier_bond.into_inner().u128(), AXL_DENOMINATION),
+                ),
+                ExecuteMsg::BondVerifier {
+                    service_name: service_name.clone(),
+                },
+            );
+            assert!(res.is_ok());
+
+            let res = execute(
+                deps.as_mut(),
+                mock_env(),
+                message_info(&api.addr_validate(verifier).unwrap(), &[]),
+                ExecuteMsg::RegisterChainSupport {
+                    service_name: service_name.clone(),
+                    chains: vec![chain_name.clone()],
+                },
+            );
+            assert!(res.is_ok());
+        }
+
+        // Create chain override with max verifiers = 3 (smaller than global max of 10)
+        let res = execute(
+            deps.as_mut(),
+            mock_env(),
+            message_info(&api.addr_make(GOVERNANCE_ADDRESS), &[]),
+            ExecuteMsg::OverrideServiceParams {
+                service_name: service_name.clone(),
+                chain_name: chain_name.clone(),
+                service_params_override: crate::msg::ServiceParamsOverride {
+                    min_num_verifiers: None,
+                    max_num_verifiers: Some(Some(3)), // Chain limit of 3
+                },
+            },
+        );
+        assert!(res.is_ok());
+
+        // Query active verifiers for the chain
+        let active_verifiers = query(
+            deps.as_ref(),
+            mock_env(),
+            QueryMsg::ActiveVerifiers {
+                service_name: service_name.clone(),
+                chain_name: chain_name.clone(),
+            },
+        )
+        .unwrap();
+
+        let active_verifiers: Vec<WeightedVerifier> = from_json(&active_verifiers).unwrap();
+
+        // Should only return 3 verifiers (respecting chain max override)
+        assert_eq!(active_verifiers.len(), 3);
+
+        // Verify they are sorted by address (since all have same weight)
+        // Should be first 3 addresses in alphabetical order
+        let mut expected_addresses = original_verifiers.clone();
+        expected_addresses.sort();
+
+        assert_eq!(
+            active_verifiers[0].verifier_info.address.to_string(),
+            expected_addresses[0]
+        );
+        assert_eq!(
+            active_verifiers[1].verifier_info.address.to_string(),
+            expected_addresses[1]
+        );
+        assert_eq!(
+            active_verifiers[2].verifier_info.address.to_string(),
+            expected_addresses[2]
+        );
```

Comment:
why is this a requirement?

Reply:
I think a reasonable requirement is that the set is stable, but I don't think we need to guarantee alphabetical order